### PR TITLE
Warn about proper validation option

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

When normalizing `pillar_roots` if the option is not a dictionary, warn about `pillar_roots` and fallback to its default.

### What issues does this PR fix or reference?

#51248 

### Previous Behavior

When validating `pillar_roots` we warned about `file_roots` and used the `file_roots` fallback.

### New Behavior

Warn about the proper option and use proper fallback for `pillar_roots`

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

File: salt/config/__init__.py [hunk 1]
```diff
@@ -1958,6 +1953,30 @@ def _validate_file_roots(file_roots):
     return file_roots
 
 
+def _validate_pillar_roots(pillar_roots):
```
Comment (waynew) [inline salt/config/__init__.py]: Does something magically call this function? I'm not seeing a change to call `_validate_pillar_roots`, so just mildly confused 😛

Comment (dwoz) [reply salt/config/__init__.py]: @waynew Good catch, fixed.

Comment (waynew) [reply salt/config/__init__.py]: Glad I'm not crazy 😀

**Review by waynew — APPROVED** [review]
👍

**Review by thatch45 — APPROVED** [review]

**Pull request merged**